lang/cs/Org.Apache.REEF.IMRU.Examples/MapperCount/IdentityMapFunction.cs

```diff
@@ -26,7 +28,7 @@ namespace Org.Apache.REEF.IMRU.Examples.MapperCount
     public sealed class IdentityMapFunction : IMapFunction<int, int>
     {
         [Inject]
-        private IdentityMapFunction()
+        private IdentityMapFunction(IInputPartition<Stream> dataPartition)
```

Comment:
In the current implementation FileSystemInputPartition of IInputPartition, when GetPartitionHandle() is called, we guarantee it is only initialize once. When we add Cache() API, we should make sure data is donwnloaded once with the same IInputPartition instance. 
Our injector for Context and Task must make sure there is only one instance of IInputPartition. Inside task, you should also check it.